glusterd2/commands/volumes/volume-smartvol-txn.go

```diff
@@ -136,3 +143,89 @@ func txnUndoPrepareBricks(c transaction.TxnCtx) error {
 
 	return nil
 }
+
+func txnCleanBricks(c transaction.TxnCtx) error {
+	var volinfo volume.Volinfo
+	if err := c.Get("volinfo", &volinfo); err != nil {
+		c.Logger().WithError(err).WithField(
+			"key", "volinfo").Debug("Failed to get key from store")
+		return err
+	}
+
+	for _, b := range volinfo.GetLocalBricks() {
+		// UnMount the Brick if mounted
+		mountRoot := strings.TrimSuffix(b.Path, b.MountInfo.Mountdir)
+		_, err := volume.GetBrickMountInfo(mountRoot)
+		if err != nil {
+			if !volume.IsMountNotFoundError(err) {
+				c.Logger().WithError(err).WithField("path", mountRoot).
+					Error("unable to get mount info")
+				return err
+			}
+		} else {
+			err := deviceutils.BrickUnmount(mountRoot)
+			if err != nil {
+				c.Logger().WithError(err).WithField("path", mountRoot).
+					Error("brick unmount failed")
+				return err
+			}
+		}
+
+		parts := strings.Split(b.MountInfo.DevicePath, "/")
+		if len(parts) != 4 {
+			return errors.New("unable to parse device path")
+		}
+		vgname := parts[2]
+		lvname := parts[3]
+		tpname, err := deviceutils.GetThinpoolName(vgname, lvname)
+		if err != nil {
+			c.Logger().WithError(err).WithFields(log.Fields{
+				"vg-name": vgname,
+				"lv-name": lvname,
+			}).Error("failed to get thinpool name")
+			return err
+		}
+
+		// Remove LV
+		err = deviceutils.RemoveLV(vgname, lvname)
```

Comment:
Just a suggestion for future: In cloud and multi-tenant environments, it's very important to wipe out data (shred or zero it out). When the same blocks are re-provisioned to some other user/customer, that user shouldn't be able to access old user's data. There are many tools that can recover it.

The wiping out can happen asynchronously/lazily in the background, and can be integrated with allocator logic using device states.